buildSrc/src/main/groovy/de/mobilej/unmock/UnMockPlugin.groovy

```diff
@@ -42,33 +42,40 @@ class UnMockPlugin implements Plugin<Project> {
 
-            if (allAndroid != null) {
-                throw new GradleException("Using 'downloadFrom' is unsupported now. Please use the unmock scope to define the android-all.jar. See https://github.com/bjoernQ/unmock-plugin/blob/master/README.md")
-            }
+        def isLib = project.plugins.findPlugin('com.android.library')
+        def isApp = project.plugins.findPlugin('com.android.application')
 
-            int unmockSize = project.configurations["unmock"].size()
-            if (unmockSize >= 1) {
-                allAndroid = project.configurations["unmock"].resolve()[unmockSize-1]
+        if(isLib || isApp) {
+            def mainVariants
+            if (isLib) {
+                mainVariants = project.android.libraryVariants
             } else {
-                throw new GradleException("Something went terribly wrong. Do a clean build and if the problem persists clear you Gradle caches.")
+                mainVariants = project.android.applicationVariants
             }
 
-            unMockTask.allAndroid = allAndroid
-            unMockTask.outputDir = project.file("${project.buildDir}/intermediates/unmock_work")
-            unMockTask.unmockedOutputJar = project.file(outputJarPath)
-            unMockTask.keepClasses = unMockExt.keep
-            unMockTask.renameClasses = unMockExt.rename
-            unMockTask.delegateClasses = unMockExt.delegateClasses
-
-            project.tasks.each {
-                task ->
-                    if (task.name ==~ /.*[cC]ompile.*/) {
-                        task.dependsOn(unMockTask)
-                    }
+            mainVariants.all { variant ->
+                 def unMockTask = project.tasks.register("unMock${variant.name.capitalize()}", UnMockTask.class) {
+                     allAndroid = project.unMock.allAndroid
+
+                     if (allAndroid != null) {
+                         throw new GradleException("Using 'downloadFrom' is unsupported now. Please use the unmock scope to define the android-all.jar. See https://github.com/bjoernQ/unmock-plugin/blob/master/README.md")
+                     }
+
+                     int unmockSize = project.configurations["unmock"].size()
+                     if (unmockSize >= 1) {
+                         allAndroid = project.configurations["unmock"].resolve()[unmockSize-1]
+                     } else {
+                         throw new GradleException("Something went terribly wrong. Do a clean build and if the problem persists clear you Gradle caches.")
+                     }
+
+                     outputDir = project.file("${project.buildDir}/intermediates/unmock_work")
+                     unmockedOutputJar = project.file(outputJarPath)
+                     keepClasses = unMockExt.keep
+                     renameClasses = unMockExt.rename
+                     delegateClasses = unMockExt.delegateClasses
+                 }
+                 variant.javaCompileProvider.configure { dependsOn unMockTask }
```

Comment:
This is depending on the compile task of the production source code, right? Is it possible to only depend on the compile test for JVM unit test code? See the change I made in my open PR to try to avoid this dependency: https://github.com/bjoernQ/unmock-plugin/pull/52 :)